src/otk/external.py

```diff
@@ -1,83 +1,73 @@
 """Support for calling external programs to transform trees. External programs
-receive the current context and the subtree to operate on in JSON. They are
-expected to return the context and subtree.
-
-Returned subtrees are resolved during tree transformation."""
+receive the subtree to operate on in JSON. They are expected to return a new
+subtree."""
 
 import json
 import logging
+import pathlib
 import subprocess
 import sys
+import os
 from typing import Any
 
-from .context import Context, OSBuildContext
+from .constant import PREFIX_EXTERNAL
 
 log = logging.getLogger(__name__)
 
 
-def call(context: Context, directive: str, tree: Any) -> Any:
-    if directive not in registry:
-        log.fatal("unknown external %r", directive)
-
-        # TODO exception
-        sys.exit(1)
-        return
-
-    required_context, executable = registry[directive]
-
-    if not isinstance(context, required_context):
-        log.fatal("external %r with wrong context %r", directive, context)
-
-        # TODO exception
-        sys.exit(1)
-        return
+def call(directive: str, tree: Any) -> Any:
+    exe = convert(directive)
```

Comment:
(super nitpick) I wonder if a slightly more specific name than `convert` would be easier to read here, maybe something like `exe = exe_from_directive(directive)` or `exe = exe_from(directive)`? (fwiw, this was marked as "done" in https://github.com/osbuild/otk/pull/94#discussion_r1625847169 but maybe got lost in some force push?)